### PR TITLE
Make book list a list of components

### PR DESCRIPTION
**Pull request by vbfox**
Instead of directly including the html elements in the parent table :

* An array of elements is exposed to react with a key (Not a really good key but it'll do)
* Each element is a separate component that will appear as such in react dev tools

**Notes:**
* Sorry for the whitespace diff, I have it in auto cleanup in vscode, I can revert
* I'll PR later to use PureComponent instead of a stateless functional one (They aren't as optimized as they can be un-pure) as it give better perf and the immutability of F# record is a perfect match
* I didn't add a key to the `Book` type but I think it should have one as it's a best practice in react anyway (And it seem that the `AzureTable` server side is reduced to using a poor `RowKey` that will create problems) .

### CODE REVIEW

File: src/Client/pages/WishList.fs [hunk 1]
```diff
@@ -243,12 +243,27 @@ let newBookForm (model:Model) dispatch =
...
                     ]
-                ]                    
-            ]        
+                ]
+            ]
         ]
     ]
 
-let view (model:Model) (dispatch: Msg -> unit) = 
+type [<Pojo>] BookProps = { key: string; book: Book; removeBook: unit -> unit }
```
Comment (forki) [inline src/Client/pages/WishList.fs]: @alfonsogarciacaro I wonder if we really need pojo everywhere?

Comment (alfonsogarciacaro) [reply src/Client/pages/WishList.fs]: For Fable 1.x unfortunately yes. If you don't use it, Fable will create a JS class for the record and React will complain as it expects plain objects.

For Fable 2.0 it won't be needed as we're planning to compile records always as plain objects :)

**Comment by forki** [comment]
I think your react patch arrived on nuget. Can you please send the second part?

**Comment by vbfox** [comment]
Yes i'll do that.

But I have a few questions on what is the exact approach for this project?
* Should I try to add a functional component somewhere else to demonstrate the feature ?
* Should I comment the code explaining WHY things are done this way so that it act as a mini-inline-tutorial ?

**Comment by forki** [comment]
whatever you think is worth tha we spread it through the ecosystem ;-)

**Pull request merged**